### PR TITLE
teensy-loader-cli: 2.1+unstable=2021-04-10 -> 2.2

### PR DESCRIPTION
**Pull request by toonn**
###### Description of changes

Bumping from an unstable revision to a tagged release.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/embedded/teensy-loader-cli/default.nix [hunk 1]
```diff
@@ -6,15 +6,15 @@
 , libusb-compat-0_1
 }:
 
-stdenv.mkDerivation rec {
+ stdenv.mkDerivation (finalAttrs: {
```
Comment (SuperSandro2000) [inline pkgs/development/embedded/teensy-loader-cli/default.nix:9]:
```suggestion
stdenv.mkDerivation {
```
we don't need to use finalAttrs here

Comment (toonn) [reply pkgs/development/embedded/teensy-loader-cli/default.nix:9]: Then how should I reuse the version attribute for `src.rev`?

**Comment by AndersonTorres** [comment]
Why github doesn't allow to go to the last line?

Anyway, remember to add a closing `)`.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
There are other two occurrences of `pname` around the code. Change them to `finalAtts.pname`.

Even better, write `teensy-loader-cli` directly.

**Comment by toonn** [comment]
Pushed. Why is hardcoding the name preferred BTW?

**Comment by AndersonTorres** [comment]
> Pushed. Why is hardcoding the name preferred BTW?

`pname` rarely changes (if it ever changes), so there is no tangible reason to use it as a parameter. 

Indeed, a change of pname is usually a fork, therefore it is easier to start an expression from scratch instead of treating them as "almost equal".

On the extreme opposite, `version` changes a lot, so it is way useful to parameterize it.

**Review by AndersonTorres — APPROVED** [review]
LGTM.
Ping me 2 days for merge.

**Pull request merged**